docker/gcp_pudl_etl.sh

```diff
@@ -107,6 +122,15 @@ copy_outputs_to_gcs
 
 # if pipeline is successful, distribute + publish datasette
 if [[ $ETL_SUCCESS == 0 ]]; then
+    if [ $GITHUB_ACTION_TRIGGER = "schedule" ]; then
+        # Update the nightly branch to point at newly successful nightly build tag
+        git config user.email "pudl@catalyst.coop"
+        git config user.name "pudlbot"
+        git remote set-url origin https://pudlbot:$PUDL_BOT_PAT@github.com/catalyst-cooperative/pudl.git
+        git checkout nightly
+        git merge --ff-only $NIGHTLY_TAG
+        git push
+    fi
     # Deploy the updated data to datasette
     if [ $GITHUB_REF = "dev" ]; then
         python ~/devtools/datasette/publish.py 2>&1 | tee -a $LOGFILE
```

Comment:
This datasette publishing should also happen whenever we finish a nightly build - not just when `$GITHUB_REF = "dev"`.

Reply:
There's a lot of fuzziness around these conditional statements, about when which actions need to happen.  I think we'll need to go over it all in more detail and make the reasoning explicit.

I think the idea here was that we wanted the nightly builds to update datasette, and they always by definition run on `dev`.  But we also wanted to be able to re-deploy Datasette with a `workflow_dispatch` if we run in on `dev` manually.